### PR TITLE
#15572 fix tabbing

### PR DESCRIPTION
**Pull request by Destrolaric**

### CODE REVIEW

File: plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/dialogs/SQLQueryParameterBindDialog.java [hunk 1]
```diff
@@ -183,21 +183,34 @@ public void widgetSelected(SelectionEvent e) {
...
-                    String newValue = ((Text) control).getText();
+                    String newValue = null;
+                    if (!(control instanceof Text)) {
+                        Control[] children = ((Composite) control).getChildren();
+                        for (Control child : children) {
+                            if (child instanceof Text) {
+                                newValue = ((Text) child).getText();
+                            }
+                        }
+                    } else {
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/dialogs/SQLQueryParameterBindDialog.java]: Why is it even a case, by the way?

Comment (Destrolaric) [reply plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/dialogs/SQLQueryParameterBindDialog.java]: When we tab out of edited cell we send composite which contains itself, button, and text. In this cases we need to find the Text child

File: plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/dialogs/SQLQueryParameterBindDialog.java [hunk 2]
```diff
@@ -183,21 +183,34 @@ public void widgetSelected(SelectionEvent e) {
...
+                    editor.addModifyListener(e -> saveEditorValue(editor, index, item));
                     editor.addTraverseListener(e -> {
                         if (e.detail == SWT.TRAVERSE_RETURN && (e.stateMask & SWT.CTRL) == SWT.CTRL) {
                             UIUtils.asyncExec(SQLQueryParameterBindDialog.this::okPressed);
                         }
                     });
-                    editor.addModifyListener(e -> saveEditorValue(editor, index, item));
-
+                    button.addTraverseListener(e -> {
+                        if (e.detail == SWT.TRAVERSE_TAB_NEXT) {
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/dialogs/SQLQueryParameterBindDialog.java]: Could we handle `SWT.TRAVERSE_TAB_PREVIOUS` as well?

File: plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/dialogs/SQLQueryParameterBindDialog.java (hunk 1 above)
Comment (ShadelessFox) [reply plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/dialogs/SQLQueryParameterBindDialog.java]: I don't see the place where we "send" our composite. The invocation of `keyTraversed` should not cause the issue IMO.

Comment (Destrolaric) [reply plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/dialogs/SQLQueryParameterBindDialog.java]: Look towards the `createEditor` method.

Comment (ShadelessFox) [reply plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/dialogs/SQLQueryParameterBindDialog.java]: I don't see anything suspicious in particular in that method.

Comment (Destrolaric) [reply plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/dialogs/SQLQueryParameterBindDialog.java]: This method is overridden inside SQLParameterBindDialog, this implementation returns not the text editor but previously mentioned composite

Comment (ShadelessFox) [reply plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/dialogs/SQLQueryParameterBindDialog.java]: Perhaps we shouldn't try to find the control of interest that way. Instead, we can bookkeep the `Text` instance as a class field and refer to it later when saving value.

**Pull request merged**